hybrid/detailed_pv_plant.py

```diff
@@ -67,6 +67,7 @@ def processed_assign(self, params):
         self.assign(params)
         self.value('system_capacity',
                    get_num_modules(self._system_model) * get_module_power(self._system_model) * 1e-3)
+        self._layout.compute_pv_layout(self.system_capacity)
```

Comment:
Why did you add this compute call to processed_assign? Logically they seem like different tasks. Normally the layout is called by `HybridLayout` but I think there's quite a bit of mess and inconsistency in that code. I'd like to straighten that up in the near future so that it's easier / more intuitive to get to the layout computation, but for now, could we just add ` pv_plant._layout.compute_pv_layout(self.system_capacity)` to the test itself?

Reply:
I think I was enforcing coherence between the parameters. Maybe instead of calling compute_pv_layout(),  _compute_string_config() and _set_system_layout() should be called separately instead?

Reply:
Hmm... `_compute_string_config` and `_set_system_layout` both require design variable inputs though, whereas the params are fixed throughout